rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromReturnNewRector.php

```diff
@@ -183,6 +184,12 @@ private function refactorDirectReturnNew(
 
         $returnType = $this->typeFactory->createMixedPassedOrUnionType($newTypes);
 
+        // skip in case of response, as handled by another rule earlier
+        /** @see ResponseReturnTypeControllerActionRector */
+        if ($returnType instanceof ObjectType && $returnType->isInstanceOf(ResponseClass::BASIC)->yes()) {
```

Comment:
The `ResponseReturnTypeControllerActionRector` only works on inside controller, if not, eg: a factory class, this seems still should be executed

Reply:
Indeed. I'll add the check

Reply:
There you go :)
https://github.com/rectorphp/rector-src/pull/6204